src/deluge/model/global_effectable/global_effectable.cpp

```diff
@@ -268,10 +276,18 @@ bool GlobalEffectable::modEncoderButtonAction(uint8_t whichModEncoder, bool on,
 				                     == RuntimeFeatureStateToggle::Off)
 				                        ? (kNumModFXTypes - 1)
 				                        : kNumModFXTypes;
-				modFXType = static_cast<ModFXType>((util::to_underlying(modFXType) + 1) % modTypeCount);
-				if (modFXType == ModFXType::NONE) {
-					modFXType = static_cast<ModFXType>(1);
+				// Get the current value to a tmp variable
+				ModFXType tmpModFXType = modFXType;
+				// Change the tmp value
+				tmpModFXType = static_cast<ModFXType>((util::to_underlying(tmpModFXType) + 1) % modTypeCount);
```

Comment:
Do we need this tmp stuff? I think it's leftover from your mod encoder button changes.